power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -33,13 +42,63 @@ inline auto build_transformer_graph(State const& /*state*/) -> TransformerGraph
     return {};
 }
 
-inline auto get_edge_weights(TransformerGraph const& /*graph*/) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
-    // TODO(mgovers): implement
-    return {};
+inline auto get_edge_weights(TransformerGraph const& graph) -> std::vector<std::pair<Idx2D, EdgeWeight>> {
+    // Step 2: Initialize the rank of all vertices (transformer nodes) as infinite (INT_MAX)
+    std::vector<EdgeWeight> rank(boost::num_vertices(graph), infty);
+    std::vector<Idx2D> sources(boost::num_vertices(graph));
+
+    // Step 3: Loop all the connected sources (status == 1)
```

Comment:
i think that this is quite obvious

```suggestion
    std::vector<EdgeWeight> rank(boost::num_vertices(graph), infty);
    std::vector<Idx2D> sources(boost::num_vertices(graph));

```

Reply:
Hmm, I do not quite follow this one

Reply:
the comments don't add new information 😬 ; for instance: `initialize the rank of all vertices as infinite` is literally the next line. might as well leave it out in that case